python/tvm/relax/frontend/pytorch.py

```diff
@@ -0,0 +1,726 @@
+            end = []
+            stride = []
+            axes = []
+            expand_dim = []
+            i = 0
+            for index in node.args[1]:
+                if isinstance(index, int):
+                    begin.append(index)
+                    end.append(index + 1)
+                    stride.append(1)
+                    axes.append(i)
+                    i = i + 1
+                elif isinstance(index, slice):
+                    begin.append(0 if index.start is None else index.start)
+                    end.append(x.shape_[i] if index.stop is None else index.stop)
+                    stride.append(1 if index.step is None else index.step)
+                    axes.append(i)
+                    i = i + 1
+                elif index is None:
+                    expand_dim.append(i)
+                else:
+                    raise ValueError("Unsupported index type: " + str(type(index)))
+            while i < len(x.shape_):
+                begin.append(0)
+                end.append(x.shape_[i])
+                axes.append(i)
+                i = i + 1
+            sliced = self.bb.emit_te(topi.strided_slice, x, begin, end, stride, axes)
+            sliced_shape = list(sliced.shape_)
+            for i in expand_dim:
+                sliced_shape.insert(i, 1)
+            return self.bb.emit(relax.op.reshape(sliced, sliced_shape))
+        else:
+            assert False
+
+    def _mul(self, node: fx.node.Node) -> relax.Var:
+        lhs, rhs = self.retrive_args(node)
+        if isinstance(lhs, relax.Var) or isinstance(rhs, relax.Var):
+            return self._call_binary_op(relax.op.multiply, lhs, rhs)
+        else:
+            return lhs * rhs
+
+    def _sin(self, node: fx.node.Node) -> relax.Var:
+        return self.bb.emit(relax.op.sin(self.env[node.args[0]]))
+
+    def _cos(self, node: fx.node.Node) -> relax.Var:
+        return self.bb.emit(relax.op.cos(self.env[node.args[0]]))
+
+    def _cat(self, node: fx.node.Node) -> relax.Var:
+        args = self.retrive_args(node)
+        return self.bb.emit(relax.op.concatenate(args[0], axis=node.kwargs["dim"]))
+
+    def _truediv(self, node: fx.node.Node) -> relax.Var:
+        lhs, rhs = self.retrive_args(node)
+        if isinstance(lhs, relax.Var) or isinstance(rhs, relax.Var):
+            return self._call_binary_op(relax.op.divide, lhs, rhs)
+        else:
+            return lhs / rhs
+
+    def _floordiv(self, node: fx.node.Node) -> relax.Var:
+        lhs, rhs = self.retrive_args(node)
+        if isinstance(lhs, relax.Var) or isinstance(rhs, relax.Var):
+            return self._call_binary_op(relax.op.floor_divide, lhs, rhs)
+        else:
+            return lhs // rhs
+
+    def _matmul(self, node: fx.node.Node) -> relax.Var:
+        args = self.retrive_args(node)
+        res = self._matmul_impl(
+            args[0],
+            args[1],
+        )
+        return res
+
+    def _matmul_impl(self, a: relax.Expr, b: relax.Expr):
```

Comment:
Need a general high-level matmul operator. Didn‘t get time to make it in the hackathon.